lint/problem.go

```diff
@@ -76,11 +76,28 @@ func (p Problem) MarshalYAML() (interface{}, error) {
 
 // Marshal defines how to represent a serialized Problem.
 func (p Problem) marshal() interface{} {
-	// The descriptor is always set, and location may be set.
-	// If they are both set, prefer the location.
-	loc := p.Location
-	if loc == nil && p.Descriptor != nil {
-		loc = p.Descriptor.GetSourceInfo()
+	var fl fileLocation
+	if p.Location != nil {
+		// If Location is set, use it.
+		fl = fileLocationFromPBLocation(p.Location, p.Descriptor)
+	} else if p.Descriptor != nil {
+		// Otherwise, use the descriptor's location.
+		// This is the protobuf-go idiomatic way to get the source location.
```

Comment:
```suggestion
		// This is the protobuf-go idiomatic way to get the source location.
		// Note: ParentFile() called on a FileDescriptor returns itself.
```

This was a question I had, and is apparent in the documentation for this API, but had me worried at first, so thought it might be useful to include to help make this all more readable.